# Análisis de regresión

## Objetivo

Construir un modelo de regresión lineal múltiple que explique el PIB per cápita a partir de variables socioeconómicas.

In [1]:
import pandas as pd
import numpy as np
import statsmodels.api as sm

df = pd.read_csv('C:/Users/Alfi/Desktop/analista de datos/estadística/proyecto estadística/proyecto-socioeconómico/data/clean/final_dataset.csv')

In [2]:
df['log_gdp'] = np.log(df['gdp_per_capita'])

In [3]:
X = df[['life_expectancy', 'internet_usage', 'unemployment']]
X = sm.add_constant(X)

y = df['log_gdp']

In [4]:
model = sm.OLS(y, X).fit()
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                log_gdp   R-squared:                       0.720
Model:                            OLS   Adj. R-squared:                  0.720
Method:                 Least Squares   F-statistic:                     3513.
Date:                Wed, 06 May 2026   Prob (F-statistic):               0.00
Time:                        10:51:44   Log-Likelihood:                -4621.7
No. Observations:                4107   AIC:                             9251.
Df Residuals:                    4103   BIC:                             9277.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                      coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------
const              -1.7730      0.152    -

## Interpretación del modelo

El modelo presenta un R² de aproximadamente 0.72, lo que indica que explica una proporción considerable de la variabilidad del PIB per cápita.

La esperanza de vida muestra el mayor coeficiente positivo, lo que sugiere que es la variable con mayor impacto en el modelo.

El uso de internet también presenta un efecto positivo y estadísticamente significativo, aunque de menor magnitud.

Por otro lado, el desempleo no resulta estadísticamente significativo, lo que indica que no aporta poder explicativo relevante cuando se consideran las demás variables.

## Limitaciones del modelo

El modelo es de naturaleza correlacional y no implica causalidad. Además, pueden existir variables omitidas y multicolinealidad entre variables independientes.